js/packages/common/src/contexts/connection.tsx

```diff
@@ -530,76 +538,23 @@ export async function sendSignedTransaction({
   timeout?: number;
 }): Promise<{ txid: string; slot: number }> {
   const rawTransaction = signedTransaction.serialize();
-  const startTime = getUnixTs();
   let slot = 0;
-  const txid: TransactionSignature = await connection.sendRawTransaction(
+
+  const txid = await sendAndConfirmRawTransaction(
+    connection,
     rawTransaction,
     {
       skipPreflight: true,
-    },
-  );
-
-  console.log('Started awaiting confirmation for', txid);
-
-  let done = false;
-  (async () => {
-    while (!done && getUnixTs() - startTime < timeout) {
-      connection.sendRawTransaction(rawTransaction, {
-        skipPreflight: true,
-      });
-      await sleep(500);
+      commitment: 'confirmed'
     }
-  })();
-  try {
-    const confirmation = await awaitTransactionSignatureConfirmation(
-      txid,
-      timeout,
-      connection,
-      'recent',
-      true,
-    );
-
-    if (!confirmation)
-      throw new Error('Timed out awaiting confirmation on transaction');
+  );
 
-    if (confirmation.err) {
-      console.error(confirmation.err);
-      throw new Error('Transaction failed: Custom instruction error');
-    }
+  const confirmation = await connection.getConfirmedTransaction(txid, 'confirmed');
 
-    slot = confirmation?.slot || 0;
-  } catch (err) {
-    console.error('Timeout Error caught', err);
-    if (typeof err === 'object' && err && 'timeout' in err) {
-      throw new Error('Timed out awaiting confirmation on transaction');
-    }
-    let simulateResult: SimulatedTransactionResponse | null = null;
-    try {
-      simulateResult = (
-        await simulateTransaction(connection, signedTransaction, 'single')
-      ).value;
-    } catch (e) {
-      console.error(e);
-    }
-    if (simulateResult && simulateResult.err) {
-      if (simulateResult.logs) {
-        for (let i = simulateResult.logs.length - 1; i >= 0; --i) {
-          const line = simulateResult.logs[i];
-          if (line.startsWith('Program log: ')) {
-            throw new Error(
-              'Transaction failed: ' + line.slice('Program log: '.length),
-            );
-          }
-        }
-      }
-      throw new Error(JSON.stringify(simulateResult.err));
```

Comment:
This functionality was nice. Like to bring back returning the program logs but with using getTransaction instead of simulate.